lib/pages/settings_page.dart

```diff
@@ -0,0 +1,111 @@
+  SettingsPage({Key key}) : super(key: key);
+
+  @override
+  SettingsPageState createState() => SettingsPageState();
+}
+
+class SettingsPageState extends State<SettingsPage> {
+  final _colorsApp = GetIt.I.get<ColorService>();
+  final _sharedPreferences = GetIt.I.get<PreferencesService>();
+  final _messagingService = GetIt.I.get<MessagingService>();
+  final _userService = GetIt.I.get<UserService>();
+
+  bool onSharedWishlist = true;
+
+  @override
+  void initState() {
+    if (_sharedPreferences.getBool("onSharedWishlist") != null) {
+      onSharedWishlist = _sharedPreferences.getBool("onSharedWishlist");
+    }
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: _colorsApp.colorTheme,
+      body: SingleChildScrollView(
+        child: SafeArea(
+          child: Column(
+            children: <Widget>[
+              Align(
+                alignment: Alignment.topLeft,
+                child: IconButton(
+                  icon: Icon(
+                    Icons.arrow_back,
+                    color: WHITE,
+                  ),
+                  onPressed: () {
+                    openProfilPage();
+                  },
+                ),
+              ),
+              Container(
+                width: MediaQuery.of(context).size.width,
+                height: MediaQuery.of(context).size.height * 0.2,
+                child: Align(
+                  alignment: Alignment.center,
+                  child: Text(
+                    getString(context, 'settings'),
+                    style: TextStyle(
+                      fontSize: 24,
+                      fontWeight: FontWeight.bold,
+                      color: WHITE,
+                    ),
+                  ),
+                ),
+              ),
+              Container(
+                width: MediaQuery.of(context).size.width * 0.9,
+                child: Row(
+                  children: [
+                    Text(
+                      getString(context, 'notifications'),
+                      style: TextStyle(
+                        fontSize: 18,
+                        fontWeight: FontWeight.bold,
+                        color: WHITE,
+                      ),
+                    ),
+                    Switch(
+                      value: onSharedWishlist,
+                      onChanged: (value) {
+                        setState(() {
+                          onSharedWishlistChange(value);
+                        });
+                      },
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> onSharedWishlistChange(bool value) async {
+    onSharedWishlist = value;
+    _sharedPreferences.setBool("onSharedWishlist", value);
+    if (value) {
+      await _messagingService.setUserAppToken(_userService.user.email);
```

Comment:
why the user interface should await the service response ?

If we di not need the result, just fire event. Remove await.